File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.c
bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize){
    bool *k;
    int i,max;
    
    k = (bool*)malloc(sizeof(bool) * candiesSize);
    max = candies[0];
    
    for(i = 1; i < candiesSize; ++i)
    {
        if(max < candies[i])
            max = candies[i];
    }
    for(i = 0; i < candiesSize; ++i)
    {
        if(candies[i] + extraCandies >= max)
            k[i] = true;
        else
            k[i] = false;
    }
    *returnSize = candiesSize;
    return k;
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> list = new ArrayList<>();
        int max = candies[0];
        
        for(int i = 1; i < candies.length; i++){
            if(candies[i] > max)
                max = candies[i];
        }
        for(int i = 0; i < candies.length; i++){
            if(candies[i] + extraCandies >= max)
                list.add(true);
            else
                list.add(false);
        }
        return list;
    }
}